cmd/podman/common/specgen.go

```diff
@@ -233,7 +233,11 @@ func FillOutSpecGen(s *specgen.SpecGenerator, c *ContainerCLIOpts, args []string
 		}
 	}
 
-	userNS := ns.UsernsMode(c.UserNS)
+	userNSMode := c.UserNS
+	if userNSMode == "" {
+		userNSMode = os.Getenv("PODMAN_USERNS")
+	}
+	userNS := ns.UsernsMode(userNSMode)
```

Comment:
This should be done in common/create.go

	createFlags.String(
		"userns", getenv("PODMAN_USERNS"),
		"User namespace to use",
	)

Reply:
Good point, fixed.